scripts/generate_syncset.py

```diff
@@ -1,8 +1,34 @@
-#!/usr/bin/python
+#!/usr/bin/env python
```

Comment:
If we don't already - it might be helpful to have input/expected output tests for this script. This is somewhat hard to parse for accuracy just from code review. And along those lines - is this shared behavior or just for this repo?